pkg/carbon/carbonassets.go

```diff
@@ -0,0 +1,171 @@
+package carbon
+
+import (
+	"embed"
+	"encoding/csv"
+	"strconv"
+	"strings"
+
+	"github.com/opencost/opencost/core/pkg/log"
+	"github.com/opencost/opencost/core/pkg/opencost"
+	"github.com/opencost/opencost/core/pkg/util"
+)
+
+//go:embed carbonlookupdata.csv
+var f embed.FS
+
+var carbonLookupNode map[interface{}]float64
+var carbonLookupDisk map[interface{}]float64
+
+// Opencost does not build network types
+var carbonValidInstanceTypes map[string]string
+var carbonValidRegions map[string]string
+
+func init() {
+
+	carbonData, err := f.ReadFile("carbonlookupdata.csv")
+	if err != nil {
+		log.Errorf("Error getting content of carbon lookup file: %s", err)
+		return
+	}
+
+	reader := csv.NewReader(strings.NewReader(string(carbonData)))
+
+	// skip header
+	_, err = reader.Read()
+	if err != nil {
+		log.Errorf("Error reading carbon lookup data: %s", err)
+		return
+	}
+
+	dat, err := reader.ReadAll()
+	if err != nil {
+		log.Errorf("Error reading carbon lookup data: %s", err)
+		return
+	}
+
+	carbonLookupNode = make(map[interface{}]float64)
+	carbonLookupDisk = make(map[interface{}]float64)
+
+	carbonValidInstanceTypes = make(map[string]string)
+	carbonValidRegions = make(map[string]string)
+
+	for _, carbonItem := range dat {
+
+		if coeff, err := strconv.ParseFloat(carbonItem[5], 64); err != nil {
+
+			log.DedupedWarningf(5, "Error setting carbon lookup item: malformed carbon cost '%s'", carbonItem[5])
+
+		} else {
+
+			provider := carbonItem[0]
+			region := carbonItem[1]
+			instanceType := carbonItem[2]
+			assetType := carbonItem[3]
+
+			switch assetType {
+			case "Node":
+				carbonLookupNode[struct {
+					provider     string
+					region       string
+					instanceType string
+				}{
+					provider:     provider,
+					region:       region,
+					instanceType: instanceType,
+				}] = coeff
```

Comment:
Using `interface{}` as the map key type and then manually defining a `struct` type every time you have to insert or look up a key isn't great. Would you please define these as proper package-level (private) structs? Further, I think there has to be a way to make the map key type not `interface{}`:
```go
type carbonLookupKeyDisk struct {
    provider string
    region string
}

var carbonLookupDisk map[carbonLookupKeyDisk]float64
```